mimic.py

```diff
@@ -0,0 +1,207 @@
+        """
+        :param other: Iterable object to concatenate
+        :return: A new MyList object that includes self and other concatenated
+        """
+        extended = self.copy()
+        extended.extend(other)
+        return extended
+
+    def _get_node(self, key: int) -> Node:
+        """
+        Find the node correspond to key
+
+        :param key: The index in list of the requested Node
+        :return: Node in the position of key
+        """
+        if key >= len(self):
+            raise IndexError('list index is out of range')
+        current = self.first_node
+        i = 0
+        while current:
+            if i == key:
+                return current
+            i += 1
+            current = current.next_node
+
+    def append(self, value):
+        if len(self) == 0:
+            self.first_node = Node(value)
+        else:
+            self._get_node(len(self) - 1).next_node = Node(value)
+        self.length += 1
+
+    def count(self, value):
+        return sum(1 for i in self if i == value)
+
+    def index(self, value):
+        for i, current_value in enumerate(self):
+            if current_value == value:
+                return i
+        raise ValueError(f'{value} is not in list')
+
+    def clear(self):
+        self.first_node = None
+        self.length = 0
+
+    def copy(self):
+        new = MyList()
+        for value in self:
+            new.append(value)
+        return new
+
+    def extend(self, other):
+        for i in other:
+            self.append(i)
+
+    def insert(self, index: int, value):
+        if index == 0:
+            self.first_node = Node(value, self.first_node)
+        else:
+            prev_node = self._get_node(index - 1)
+            prev_node.next_node = Node(value, prev_node.next_node)
+        self.length += 1
+
+    def pop(self, index: int):
+        if index == 0:
+            pop_node = self.first_node
+            self.first_node = self.first_node.next_node
+        else:
+            prev_node = self._get_node(index - 1)
+            pop_node = prev_node.next_node
+            prev_node.next_node = pop_node.next_node
+        self.length -= 1
+        return pop_node.data
+
+    def remove(self, value):
+        index = self.index(value)
+        self.pop(index)
+
+    def reverse(self):
+        new = MyList()
+        for i in range(len(self) - 1, -1, -1):
+            new.append(self[i])
+        self.first_node = new.first_node
+
+    def sort(self):
+        if len(self) == 0:
+            return
+        swap_occurred = True
+        current_node = self.first_node
+        while swap_occurred:
+            swap_occurred = False
+            while current_node.next_node:
+                next_node = current_node.next_node
+                if current_node.data > current_node.next_node.data:
+                    swap_occurred = True
+                    current_node.data, current_node.next_node.data = current_node.next_node.data, current_node.data
```

Comment:
Try to avoid using double assignments.
It's very hard to read, and not really necessary.